modules/social_features/social_profile/src/SocialProfileRenderService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\social_profile;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+
+/**
+ * Class SocialProfileRenderService.
+ *
+ * @package Drupal\social_profile
+ */
+class SocialProfileRenderService {
```

Comment:
We should add and implement an interface for this service so that it can be used instead of the class. This allows projects building on Open Social to create alternative implementations if they need to.

See #1382 for an example of this. Specifically commit 538c6ca